### PR TITLE
Feature/api calls

### PR DESCRIPTION
**Pull request by GrantXBeard**
<h1>PULL REQUEST</h1>

<ul>
  <li>What does it change?</li>
Adds fetch calls and displays API data.
  <li>What does it fix?</li>
  <li>Is this a bug or a feature?</li>
feature
  <li> Does it break any existing functionality or force me to update to a new version?</li>
No
  <li>How has it been tested?</li>
</ul>

<h2>Type of Change</h2>

_Delete options that are not relevant_

- [ ] **Bug Fix**
- [x] **New Feature**
- [ ] **New Change**

<h2>How has this been tested?</h2>

- [ ] **JS**
- [ ] **Dev Tools**
- [ ] **Website**

<h2>Checklist</h2>

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new bugs
- [x] I have tagged project partners for review & it has been reviewed 

<h2>Does this PR close any issues?</h2>
#6

### CODE REVIEW

**Comment by abodnar1** [comment]
Looks good Grant! The changes were minimal but packed a punch.

File: src/App.js [hunk 1]
```diff
@@ -8,15 +8,55 @@ class App extends Component {
...
+      error: '',
     }
   }
 
-  displayMovieDetails = (id) => {
-    const singleMovie = this.state.movies.find(movie => movie.id === id)
+  componentDidMount = () => {
+    this.setState({isLoading: true})
+
+    fetch("https://rancid-tomatillos.herokuapp.com/api/v2/movies")
```
Comment (Kalikoze) [inline src/App.js:21]: Nice work setting up your network request in `componentDidMount`.  A nice refactor could be to move the network request into a `apiCalls.js` file and import it/call it here!

File: src/App.js [hunk 2]
```diff
@@ -8,15 +8,55 @@ class App extends Component {
...
+      .then(res => {
+        if (res.ok) {
+          return res.json()
+        } else {
+          console.log("Error")
+        }
```
Comment (Kalikoze) [inline src/App.js:27]: Nice job checking that the response is okay before parsing it.  Make sure to **throw a new Error** though so it hits the catch as opposed to logging 'Error'.

File: src/MovieDetail.js [hunk 3]
```diff
@@ -11,14 +11,14 @@ const MovieDetail = ({ details , closeMovieDetails }) => {
...
       <div className="copy-wrapper">
         <div className="copy">
-          <p>Tagline</p>
-          <p>Overview</p>
-          <p>Genres</p>
-          <p>Runtime</p>
+          <p>"{details.tagline}"</p>
+          <p>{details.overview}</p>
+          <p>Genres: {details.genres.join(', ')}</p>
+          <p>{details.runtime} min.</p>
```
Comment (Kalikoze) [inline src/MovieDetail.js:17]: If you wanted to, you could do another layer of destructuring here.  Something like:

```
const { tagline, overview, genres, runtime } = details;

return (
<div>
  <p>{tagline}</p>
  <p>{overview}</p>
</div>
)
```

Let me know if that makes sense or if you have any additional questions!

File: src/App.js [hunk 4]
```diff
@@ -29,6 +69,8 @@ class App extends Component {
         <header className="App-header">
           <h1>Rancid Tomatillos</h1>
         </header>
+          {this.state.isLoading && <p>Loading...</p>}
+          {this.state.error && <h2>{this.state.error}</h2>}
```
Comment (Kalikoze) [inline src/App.js:73]: Great use of conditional rendering here!

**Review by Kalikoze — COMMENTED** [review]
Nice work ya'll!  Just make sure to be something other than console.logging the error if the `response.ok` property is not true!

**Pull request merged**